tests/self_peertest.rs

```diff
@@ -8,43 +8,24 @@ mod test {
         tracing_subscriber::fmt::init();
 
         // Run a client, as a buddy peer for ping tests, etc.
-        let (buddy_enode, buddy_client_exiter) = {
-            let trin_config = TrinConfig::new_from(
-                [
-                    "trin",
-                    "--internal-ip",
-                    "--unsafe-private-key",
-                    "7261696e626f7773756e69636f726e737261696e626f7773756e69636f726e73",
-                ]
-                .iter(),
-            )
-            .unwrap();
-            let web3_ipc_path = trin_config.web3_ipc_path.clone();
-            let exiter = trin::run_trin(trin_config, String::new()).await.unwrap();
-            let enode = peertest::jsonrpc::get_enode(&web3_ipc_path).unwrap();
-            (enode, exiter)
-        };
+        let (buddy_enode, buddy_client_exiter) = peertest::launch_buddy_node().await;
 
-        let peertest_config =
-            peertest::PeertestConfig::new_from([".", "--target-node", &buddy_enode].iter())
-                .unwrap();
+        let peertest_config = peertest::PeertestConfig::new_from(["."].iter()).unwrap();
 
         // Run a client, to be tested
         let trin_config = TrinConfig::new_from(
```

Comment:
If we refactor `launch_buddy_node` as suggested above, we can use it here too

Reply:
Hm, I'm not really sold on this generalization of `launch_buddy_node`. I think `run_trin()` is the general one-liner function, already. For one example: I want the buddy peer to have a consistent node id during testing, but don't want that same node id for the node under test. So I guess we'd have to make that an argument too. At some point, we're just recreating the whole run_trin() interface, passing in the config in a slightly different way.